SYCL/ESIMD/api/functional/functions/functions_select_2d.hpp

```diff
@@ -0,0 +1,286 @@
+  static constexpr int StrideX = StrideXT::value;
+  static constexpr int OffsetX = OffsetXT::value;
+  static constexpr int OffsetY = OffsetYT::value;
+  static constexpr int Height = NumElems / SimdSplitCoeffT::value;
+  static constexpr int Width = NumElems / Height;
+  using TestDescriptionT = TestDescription<NumElems, TestCaseT>;
+
+public:
+  bool operator()(sycl::queue &queue, const std::string &data_type) {
+    bool passed = true;
+
+    static_assert(NumElems == Height * Width,
+                  "Invalid SimdSplitCoeff, height and width multiplacation "
+                  "should be equal to simd size.");
+    static_assert(Width > SizeX);
+    static_assert(Height > SizeY);
+    static_assert(Height > OffsetY);
+    static_assert(Width > OffsetX);
+
+    constexpr size_t value_for_increase_ref_data_for_change = 50;
+    static_assert(std::numeric_limits<signed char>::max() >
+                      value_for_increase_ref_data_for_change + NumElems,
+                  "Value that used for increase ref data for fill plus simd "
+                  "size should be less than char max value.");
+
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> result(NumElems, allocator);
+    shared_vector<DataT> initial_ref_data(NumElems, allocator);
+    shared_vector<DataT> ref_data_for_change(SizeY * SizeX, allocator);
+
+    std::iota(initial_ref_data.begin(), initial_ref_data.end(), 1);
+    // We should have different values in the first reference data and in the
+    // second reference data.
+    std::iota(ref_data_for_change.begin(), ref_data_for_change.end(),
+              initial_ref_data.back() + value_for_increase_ref_data_for_change);
+
+    queue.submit([&](sycl::handler &cgh) {
+      DataT *init_ref_ptr = initial_ref_data.data();
+      DataT *ref_data_for_change_ptr = ref_data_for_change.data();
+      DataT *const out_ptr = result.data();
+      cgh.single_task<
+          Kernel<DataT, NumElems, TestCaseT, SizeYT, StrideYT, SizeXT, StrideXT,
+                 SimdSplitCoeffT, OffsetXT, OffsetYT>>([=]() SYCL_ESIMD_KERNEL {
+        TestCaseT::template call_esimd_function<DataT, NumElems, SizeY, StrideY,
+                                                SizeX, StrideX, Height, Width>(
+            init_ref_ptr, ref_data_for_change_ptr, out_ptr, OffsetX, OffsetY);
+      });
+    });
+    queue.wait_and_throw();
+
+    std::vector<size_t> selected_indexes;
+    // Collect the indexess that has been selected.
+    for (int i = 0; i < SizeY; ++i) {
+      for (int j = 0; j < SizeX; ++j) {
+        selected_indexes.push_back(OffsetY * Width + i * Width * StrideY +
+                                   j * StrideX + OffsetX);
+      }
+    }
+
+    // Push the largest value to avoid the following error: can't dereference
+    // out of range vector iterator.
+    selected_indexes.push_back(std::numeric_limits<size_t>::max());
+    auto selected_indexses_ptr = selected_indexes.begin();
```

Comment:
```suggestion
    auto next_selected_index = selected_indexes.begin();
```
Let's stick with the `next_selected_index` you've used for `1d select` here: https://github.com/intel/llvm-test-suite/pull/875#discussion_r824771178 - that's iterator after all.

Reply:
I've applied this comment in this commit: https://github.com/intel/llvm-test-suite/pull/908/commits/cfdb4c12f261d96d9fdc262394834f02be264247